@xen-orchestra/lite/src/stores/modal.store.ts

```diff
@@ -12,9 +12,11 @@ import {
 
 export const useModalStore = defineStore("modal", () => {
   const modals = ref(new Map<symbol, ModalController>());
+  const closeEvent = createEventHook();
```

Comment:
This event hook should not be created inside the store root. When doing that, it will be shared across all modals.

I think we should move the `closeEvent` and the `close` function inside the `open` function (and remove it from the returned object):

```ts
const approveEvent = createEventHook<T>();
const declineEvent = createEventHook();
const closeEvent = createEventHook();

const close = async () => {
  await closeEvent.trigger(undefined);
  modals.value.delete(id);
};
```

Then replace `close(id)` in both `approve` and `decline` functions with `await close()`